chain-core/src/init/coin.rs

```diff
@@ -5,23 +5,22 @@
 
 use crate::init::{MAX_COIN, MAX_COIN_DECIMALS, MAX_COIN_UNITS};
 use crate::state::tendermint::TendermintVotePower;
+use crate::state::tendermint::TENDERMINT_MAX_VOTE_POWER;
 use parity_codec::{Decode, Encode, Input};
 
-use crate::state::tendermint::TENDERMINT_MAX_VOTE_POWER;
-#[cfg(feature = "serde")]
-use serde::de::{Deserialize, Deserializer, Error, Visitor};
-#[cfg(feature = "serde")]
-use serde::Serialize;
+use serde::{Deserialize, Serialize, Serializer};
+
+use serde::de::{Deserializer, Error, Visitor};
+
 use static_assertions::const_assert;
 use std::convert::TryFrom;
 use std::{fmt, mem, ops, result, slice};
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Encode)]
-#[cfg_attr(feature = "serde", derive(Serialize))]
 pub struct Coin(u64);
 
 /// error type relating to `Coin` operations
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, serde::Serialize, Deserialize)]
```

Comment:
Why do we have to implement `Serialize` and `Deserialize` for `CoinError`?
```suggestion
#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
```

Reply:
if not, compiling fails

Reply:
Strange. I don't see any reason for compilation to fail.

Reply:
yes, it compiles fine. i'll remove it.